src/web/web-service/ot_client.cpp

```diff
@@ -0,0 +1,205 @@
+#include "ot_client.hpp"
+
+#include <openthread/platform/toolchain.h>
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#include "platform-posix.h"
+#include "common/code_utils.hpp"
+#include "common/logging.hpp"
+
+namespace ot {
+
+Client::Client(void)
+    : mTimeout(kDefaultTimeout)
+    , mSocket(-1)
+{
+}
+
+Client::~Client(void)
+{
+    if (mSocket != -1)
+    {
+        close(mSocket);
+    }
+}
+bool Client::Connect(void)
+{
+    struct sockaddr_un sockname;
+    int                ret;
+
+    mSocket = socket(AF_UNIX, SOCK_STREAM, 0);
+    VerifyOrExit(mSocket != -1, perror("socket"); ret = OT_EXIT_FAILURE);
+
+    memset(&sockname, 0, sizeof(struct sockaddr_un));
+    sockname.sun_family = AF_UNIX;
+    strncpy(sockname.sun_path, OPENTHREAD_POSIX_APP_SOCKET_NAME, sizeof(sockname.sun_path) - 1);
+
+    ret = connect(mSocket, reinterpret_cast<const struct sockaddr *>(&sockname), sizeof(struct sockaddr_un));
+
+    if (ret == -1)
+    {
+        otbrLog(OTBR_LOG_ERR, "OpenThread daemon is not running.");
+    }
+
+exit:
+    return ret == 0;
+}
+
+char *Client::Execute(const char *aFormat, ...)
+{
+    va_list args;
+    int     ret;
+    char *  rval = NULL;
+    ssize_t count;
+    size_t  rxLength = 0;
+
+    va_start(args, aFormat);
+    ret = vsnprintf(&mBuffer[1], sizeof(mBuffer) - 1, aFormat, args);
+    va_end(args);
+
+    if (ret < 0)
+    {
+        otbrLog(OTBR_LOG_ERR, "Failed to generate command: %s", strerror(errno));
+    }
+
+    mBuffer[0] = '\n';
+    ret++;
+
+    if (ret == sizeof(mBuffer))
+    {
+        otbrLog(OTBR_LOG_ERR, "Command exceeds maximum limit: %d", kBufferSize);
+    }
+
+    mBuffer[ret] = '\n';
+    ret++;
+
+    count = write(mSocket, mBuffer, ret);
+
+    if (count < ret)
+    {
+        mBuffer[ret] = '\0';
+        otbrLog(OTBR_LOG_ERR, "Failed to send command: %s", mBuffer);
+    }
+
+    for (int i = 0; i < mTimeout; ++i)
```

Comment:
Why not select with the total timeout?

Reply:
`select()` will return the moment the fd is readable, so read() may not receive all outputs. Especially for `scan`